test/externalissuer/external_issuer_test.go

```diff
@@ -0,0 +1,156 @@
+package externalissuer
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var TestHelper *testutil.TestHelper
+
+const (
+	TestAppBackendDeploymentName = "backend"
+	TestAppNamespaceSuffix       = "external-issuer-app-test"
+)
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+func TestInstallApp(t *testing.T) {
+	if !TestHelper.ExternalIssuer() {
```

Comment:
Not really, Currently its going to get picked up when we do: 

```
function deep_integration_tests() {
    run_test "$test_directory/install_test.go" --linkerd-namespace=$linkerd_namespace
    exit_on_err "error during install"

    run_test "$(go list $test_directory/.../...)" --linkerd-namespace=$linkerd_namespace
    exit_on_err "error during deep tests"
}
```

Maybe we can restructure directories to have a more clear separation of test categories. Wdyt  @alpeb  ?

Reply:
@zaharidichev not sure about that, what do you have in mind?
Also, I just realized you can avoid repeating `if !TestHelper.ExternalIssuer()` in each function by dealing with that inside `TestMain`. Perhaps extracting and exposing the exiting logic inside `NewTestHelper` to reuse that.

Reply:
@alpeb What I had in mind is to put the external issuer into a separate test directory so it will not be caught by `run_test "$(go list $test_directory/.../...)` which is used in `function deep_integration_tests()`. But yes, not sure about that either. Simply extracting the logic into TestHelper also works: 

```scala
func (h *TestHelper) SkipIfNotConfiguredForExternalIssuer(t *testing.T) {
	if !h.ExternalIssuer() {
		t.Skip("Skiping as --external-issuer=false")
	}
}
```

Is that what you had in mind ?

Reply:
Moving the test into a separate directory will still require having guards to avoid having it run when running unit tests `go test ./...` from the root. IMO the current layout is fine for now.

My other comment was to actually get rid of all those ifs in all the tests in that file, and just have one in `TestMain`. Just like if `testutil.NewTestHelper()` in `TestMain` detects `- -integration-tests` wasn't used then none of the test functions in the file are executed, you could add the check for `--external-issuer` in there as well.

Reply:
Ahhh.. now I see what you mean :)